mmv1/third_party/terraform/provider/provider_internal_test.go

```diff
@@ -688,3 +689,272 @@ func TestProvider_providerConfigure_project(t *testing.T) {
 		})
 	}
 }
+
+func TestProvider_providerConfigure_requestTimeout(t *testing.T) {
+	cases := map[string]struct {
+		ConfigValues        map[string]interface{}
+		ExpectedValue       string
+		ExpectedSchemaValue string
+		ExpectError         bool
+		ExpectFieldUnset    bool
+	}{
+		"Invalid timeout value set": {
+			ConfigValues: map[string]interface{}{
+				"request_timeout": "timeout",
+			},
+			ExpectedValue:       "timeout",
+			ExpectedSchemaValue: "timeout",
+			ExpectError:         true,
+			ExpectFieldUnset:    false,
+		},
+		"when config is unset, the value will be 0s in order to set the default value": {
+			ExpectedValue:    "0s", // TODO: check why default value is not 30s
```

Comment:
We know that the default value is set elsewhere now, so you can remove the TODOs :)

Instead, could you add some short comments explaining why the test shows the default value is not present? This could be link to a GitHub issue, or permalinks to other lines of code. What do you think would be most helpful to someone reading this for the first time?